app/bundles/ReportBundle/Event/ColumnCollectEvent.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\ReportBundle\Event;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+class ColumnCollectEvent extends Event
+{
+    private string $object;
+
+    /**
+     * @var array<string, mixed>
+     */
+    private array $properties;
+
+    /**
+     * @var array<string, mixed>
+     */
+    private array $columns;
+
+    /**
+     * @param array<string, mixed> $properties
+     */
+    public function __construct(string $object, array $properties = [])
+    {
+        $this->object     = $object;
+        $this->properties = $properties;
+        $this->columns    = [];
+    }
```

Comment:
Just some syntax sugar we can use when we've dropped PHP 7.4
```suggestion
    /**
     * @var array<string, mixed>
     */
    private array $columns;

    /**
     * @param array<string, mixed> $properties
     */
    public function __construct(private string $object, private array $properties = [])
    {
        $this->columns = [];
    }
```